### ISSUE TITLE
NVDA doesn't read state "readonly" when users focused into the spinbutton of the web.

### ISSUE THREAD

**Issue by a11y-nvisions** [nvaccess/nvda#13672]
### Steps to reproduce:
- Go to the [Test Page](https://a11y-nvisions.github.io/BugReports/nvda/spinReadonly/index.html),
- Three read-only inputs is there, Navigate them pressing a Tab key.
- Each of three inputs is the different text-edtiable type or tag, the last of them is spin button.
- When you focus on the last spin button field, make sure that the NVDA announces "Read Only".

### Actual behavior:
NVDA doesn't report a "Read Only".NVDA is speaking "Spin Button Input, Spin Button, Editable".

### Expected behavior:
NVDA should speech like this, "Spin Button Input, Spin Button, Read Only, Blank".

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2021.3.5

#### Windows version:
Windows 11 21H2 (OS Build 22000.613)

#### Name and version of other software in use when reproducing the issue:
- Firefox 100
- Chrome 100

#### Other information about your system:
None

#### Does the issue still occur after restarting your computer?
I didn't try it. but, I'm sure Restarting won't affect at speech.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No, tested on only 2021.3.5.

#### If NVDA add-ons are disabled, is your problem still occurring?
I didn't try it, too.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
I didn't try it, too. and I think It's not related.

**Comment by XLTechie** [comment]
I can replicate this.

Do you have any examples of this happening in the wild?

**Comment by a11y-nvisions** [comment]
Hello, @XLTechie 
Yes, We saw it while working. We're a team working that give solutions, counseling it for web accessibility to our clients.

We found out it from our regular client's shopping mall page.
This page has product lists, the custom spin buttons are by each item of this list.

custom spin button consists the below three elements.
- decreasement button for removing a product from cart
- Increasement button for adding a product to cart
- Read-only spin button for displaying product quantity in the cart
- 
But. Unfortunately, we can't give you their page link for security, we're sorry.

**Comment by seanbudd** [comment]
The behaviour seems to be the same in Chrome and Firefox, suggesting that this is an issue with NVDA. It appears that both "editable" (IAccessible2 states) and "readonly" (accState) attributes are added to the object. We need to investigate how both these states are added.

Dev info for the spin control using Chrome:

```py
IO - inputCore.InputManager.executeGesture (16:52:55.898) - winInputHook (22456):
Input: kb(desktop):NVDA+f1
IO - speech.speech.speak (16:52:55.912) - MainThread (15240):
Speaking [LangChangeCommand ('en_GB'), 'NVDA+f1']
DEBUG - gui.contextHelp.bindHelpEvent (16:52:55.949) - MainThread (15240):
Did context help binding for LogViewer
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (16:52:56.141) - MainThread (15240):
Developer info for navigator object:
name: 'SpinButton input'
role: Role.SPINBUTTON
roleText: None
states: State.READONLY, State.FOCUSABLE, State.FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.IAccessible.ia2Web.Editor object at 0x082DB250>
Python class mro: (<class 'NVDAObjects.IAccessible.ia2Web.Editor'>, <class 'NVDAObjects.IAccessible.ia2Web.Ia2Web'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.DocumentWithTableNavigation'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: None
location: RectLTWH(left=293, top=579, width=365, height=48)
value: ''
appModule: <'appModuleHandler' (appName 'chrome', process ID 13028) at address 57aebf0>
appModule.productName: 'Google Chrome'
appModule.productVersion: '101.0.4951.64'
TextInfo: <class 'NVDAObjects.IAccessible.ia2TextMozilla.MozillaCompoundTextInfo'>
windowHandle: 2230480
windowClassName: 'Chrome_RenderWidgetHostHWND'
windowControlID: 999641952
windowStyle: 1445986304
extendedWindowStyle: 32
windowThreadID: 13108
windowText: 'Chrome Legacy Window'
displayText: ''
IAccessibleObject: <POINTER(IAccessible2) ptr=0x50f0044 at 6ebc940>
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=2230480, objectID=-4, childID=-988242
IAccessible accName: 'SpinButton input'
IAccessible accRole: ROLE_SYSTEM_SPINBUTTON
IAccessible accState: STATE_SYSTEM_FOCUSED, STATE_SYSTEM_READONLY, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048644)
IAccessible accDescription: None
IAccessible accValue: ''
IAccessible2 windowHandle: 2230480
IAccessible2 uniqueID: -988242
IAccessible2 role: ROLE_SYSTEM_SPINBUTTON
IAccessible2 states: IA2_STATE_EDITABLE, IA2_STATE_OPAQUE, IA2_STATE_SELECTABLE_TEXT, IA2_STATE_SINGLE_LINE (13320)
IAccessible2 attributes: 'display:block;tag:input;explicit-name:true;id:spin_edit;text-align:left;text-model:a1;text-input-type:number;'
IAccessible2 relations: labelledBy
```

**Comment by Sureshkumar2288** [comment]
Hi All, when we try to read the NVDA as readonly state, the readonly state does not read as NVDA other than the role as textbox element. please suggest how can we solve this issue

**Comment by Adriani90** [comment]
Yeah it seems NVDA renders only text areas and input text as read only correctly, but not numbers. Also spin_edit readonly is not included in NVDA cc: @jcsteh there is also an aria readonly atribute but I think this refers to different controls right?

**Comment by msftedad** [comment]
Hi Team any update on ticket as I can see #14827 is closed now so please work and update status here on this ticket.

**Comment by jcsteh** [comment]
Read only text controls are a bit of a tricky case. It's easiest to explain this by example. Consider a terms of service text box like this:
`<textarea readonly>`
NVDA reports this as "edit read only". That seems contradictory, but what is being indicated here is that this has all the functionality of a normal edit box (cursor keys, etc.), but it is read only. You see this even in desktop applications and other screen readers do this as well.

Now let's consider the read only combo box from #14827:
`<input role="combobox" readonly>`
There are two types of combo boxes: combo boxes that allow you to enter a value (including cursoring through that value with the left and right arrow keys) and combo boxes that only pop up a list. This one is somewhere in the middle. Because it's an input, that means it has text box functionality; you can cursor through its value, even if it's read only. This is in contrast to a `<select>`, where you can't. I'm not really sure what NVDA should read there. It should definitely report read only, but it shouldn't just ignore the navigable aspect altogether. For now, the closest we can probably get is "combo box editable read only", but that's kinda confusing.

**Comment by Adriani90** [comment]
@jcsteh what about something like "spin button not adjustable" or "combo box not adjustable"?
